tests/net/util.rs

```diff
@@ -0,0 +1,76 @@
+//! Utility functions
+//!
+//! A collection miscellaneous functions that are used in the tests, but are generic enough to be
+//! factored out.
+
+use std::ops;
+
+/// Subslicing.
+pub trait SubSlice
+where
+    Self: ops::Index<ops::Range<usize>>,
+{
+    /// Create new subslice of given size or smaller.
+    ///
+    /// Functions similar to `&sl[a..b]`, but while regular slicing will panic if `b` is out of
+    /// range, `subslice` will return the longest possible slice.
+    #[inline]
+    fn subslice(
+        &self,
+        range: ops::Range<usize>,
+    ) -> &<Self as ops::Index<ops::Range<usize>>>::Output;
+}
+
+impl<T> SubSlice for [T] {
+    #[inline]
+    fn subslice(
+        &self,
+        mut range: ops::Range<usize>,
+    ) -> &<Self as ops::Index<ops::Range<usize>>>::Output {
+        if range.end > self.len() {
+            range.end = self.len();
+        }
```

Comment:
Should the case `range.start > self.len()` also be taken into account?

Reply:
Fixed, also added a test.